aidial_assistant/json_stream/json_parser.py

```diff
@@ -38,34 +46,43 @@ def string_node(node: JsonNode) -> JsonString:
     return node
 
 
-class JsonParser:
-    @staticmethod
-    @asynccontextmanager
-    async def parse(stream: CharacterStream) -> AsyncGenerator[JsonNode, Any]:
-        root = JsonRoot()
+class RootNodeResolver(NodeResolver):
```

Comment:
It's courious that in the module called `json_parser` the main classes aren't called like parsers. I suggest:

`RootNodeResolver` -> `RootNodeParser`

`NodeResolver` -> `NodeParser`

`NodeResolver.resolve` -> `NodeParser.parse`